### DISCUSSION TITLE
Script to add multiple blacklist not working on v0.107.15

### DISCUSSION THREAD

**Discussion by vannimb** [AdguardTeam/AdGuardHome#4997] [General]
Hello Everyone,
i was using teh script at this link to add multiple blacklists to every new adguard instance i am working with.
here the link:
https://gist.githubusercontent.com/LeiHao0/4a4235df0fb97d559f9876eab0ebb5a4/raw/68ba302714cf60655185f0c999c5dc50f6a1882c/gistfile1.py

unfortunatley it looks not to work anylonger and i get  this kind of output:
{"url": "https://urlhaus.abuse.ch/downloads/hostfile/", "name": "https://urlhaus.abuse.ch/downloads/hostfile/", "whitelist": false}
Forbidden

Could someone help me to fix the issue and be able to add multiple blacklists in a shot?
Best regards

**Comment by ameshkov** [comment]
Please check the API changes, multiple methods now require application/json:
https://github.com/AdguardTeam/AdGuardHome/blob/master/CHANGELOG.md#v010714---2022-09-29

**Reply by vannimb** [reply]
Thanks for the quick answer @ameshkov !
Sorry for my ignorance, i know is not realted to Adguard but... do you have any idea how to modify the script to make it work with the new adguard version?
Best regards

**Reply by ainar-g** [reply]
@vannimb, make sure that you set `Content-Type` to `application/json`.

**Reply by vannimb** [reply]
Many thanks for your recommendation, it was enough to modify the following line as:
headers = {'User-Agent': 'Mozilla/5.0 (Windows NT 6.1; WOW64; rv:52.0) Gecko/20100101 Firefox/52.0'**, 'Content-Type':'application/json'**}

best regards

**Reply by ainar-g** [reply]
You don't really need a User-Agent there.  Also, how do you authenticate?  It's not clear from the script, if it adds the userinfo section (`user:password@`) to the URL or not.

**Comment by vannimb** [comment]
just a comment:
more in general, it could be anew nice feature to have the chance to add multiple blacklists from the gui and not having to use 3rd party scripts. :) (but maybe is not the right formu section)
